delphi/translators/for2py/pyTranslate.py

```diff
@@ -633,22 +666,25 @@ def printIf(self, node, printState: PrintState):
             )
 
     def printOp(self, node, printState: PrintState):
-        expr_str = self.proc_expr(node, False)
+        expr_str = self.proc_expr(node, False, printState)
         self.pyStrings.append(expr_str)
 
     def printLiteral(self, node, printState: PrintState):
-        expr_str = self.proc_literal(node)
+        expr_str = self.proc_literal(node, printState)
         self.pyStrings.append(expr_str)
 
+
     def printRef(self, node, printState: PrintState):
-        ref_str = self.proc_ref(node, False)
+        ref_str = self.proc_ref(node, False, printState)
         self.pyStrings.append(ref_str)
 
     def printAssignment(self, node, printState: PrintState):
+
         assert len(node["target"]) == 1 and len(node["value"]) == 1
         lhs, rhs = node["target"][0], node["value"][0]
 
-        rhs_str = self.proc_expr(node["value"][0], False)
+        printState.isAssignment = True
```

Comment:
I'm trying to understand why `printState` has been added as an argument to `proc_expr()`, `proc_ref()`, etc.  It has the effect of weaving the entire state of this computation through the entire computation, and makes it harder to understand and maintain the code.  I would like to avoid this if at all possible.

Is `printState` being used just to indicate whether an expression is being used on the RHS of an assignment via the `isAssignment` flag?  If so, it seems simpler to just pass down a boolean argument indicating this.